### PR TITLE
[APPS-10910] FullScreenLoading component

### PR DESCRIPTION
**Pull request by martin-eq**
### Changes
Migrating `FullScreenLoading` component from react-common to react-common-ts.

### Tasks ([Engineering planning document](https://docs.google.com/spreadsheets/d/111MLafVCwaZyEPExSMkg7TTBHBKDjizgjWdLjlbsh80))
- [x] Code complete

### JIRA Ticket
https://ipsycorp.atlassian.net/browse/APPS-10910

### Related people
\cc @matiasherranz @jupadev

### CODE REVIEW

File: declarations.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+declare module 'my-config' {
+  global {
+    interface Document {
+      documentMode?: number;
+    }
+  }
+}
+
+// Declare a Nullable type for convenience
+type Nullable<T> = T | null;
```
Comment (jupadev) [inline declarations.d.ts:17]: 🥇

File: src/components/FullScreenLoading/index.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ *   non-animated one. In this case, special CSS classes are used to animate
+ *   the image and make it slide from right to left.
+ */
+
+import rolling from './rolling.svg';
+import heart from './heart.gif';
+
+const IMAGE_TYPE_ANIMATED = 'ANIMATED';
+
+type defaultPropTypes = {
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:24]: it should be capitalized `DefaultPropTypes`

File: src/components/FullScreenLoading/index.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  loading?: boolean;
+  children?: ReactNode;
+  opacity?: number;
+  zIndex?: number;
+  spinnerWidth?: number;
+  message?: ReactNode | string;
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:30]: `ReactNode` already include `string`
image

File: src/components/FullScreenLoading/index.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+};
+
+type TimeoutID = number;
+
+export const DefaultFullScreenLoading = ({
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:35]: `FC<DefaultPropTypes >`

File: src/components/FullScreenLoading/index.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      {children}
+      <div className="full-screen-loading" style={{ zIndex }}>
+        <img src={isIE11 ? heart : rolling} alt="loading" style={style} />
+        <p className="full-screen-loading-message">{message}</p>
+      </div>
+    </div>
+  );
+};
+
+type propTypes = defaultPropTypes & {
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:74]: same above comment

File: src/components/FullScreenLoading/index.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  imageOverride?: string;
+  imageOverrideType?: string;
+  timeout?: number;
+};
+
+const FullScreenLoading = (props: propTypes): ReactElement => {
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:80]: same above comment

File: src/components/FullScreenLoading/index.tsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    imageType === IMAGE_TYPE_ANIMATED;
+
+  useEffect(() => {
+    if (props.timeout) {
+      timeoutId = setTimeout(hideInterstitial, props.timeout);
+    }
+
+    return () => {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:107]: 👍

File: src/components/FullScreenLoading/index.tsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+          />
+        </div>
+      </div>
+    );
+  }
+
+  return <div>{markup}</div>;
+};
+
+FullScreenLoading.defaultProps = {
```
Comment (jupadev) [inline src/components/FullScreenLoading/index.tsx:168]: we can use the default props within component.

`
const Component: FC<Props> = ( {
    imageOverride: undefined,
    imageOverrideType: undefined,
    opacity = 1,
    zIndex,
    spinnerWidth,
    message,
    children,
    loading,
   ..rest
  } ) => {
return null
}
`

Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Addressed in 375bee3

File: src/components/FullScreenLoading/index.tsx (hunk 6 above)
Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Addressed in 375bee3

File: src/components/FullScreenLoading/index.tsx (hunk 5 above)
Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Addressed in 375bee3

File: src/components/FullScreenLoading/index.tsx (hunk 4 above)
Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Addressed in 375bee3

File: src/components/FullScreenLoading/index.tsx (hunk 3 above)
Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Great catch! Thank you. Addressed in 375bee3

File: src/components/FullScreenLoading/index.tsx (hunk 2 above)
Comment (martin-eq) [reply src/components/FullScreenLoading/index.tsx]: Addressed in 375bee3

**Pull request merged**